### PR TITLE
Fixes #31023 - add katello_agent_enabled macro for provisioning

### PR DESCRIPTION
**Pull request by jturel**

### CODE REVIEW

File: app/lib/katello/concerns/base_template_scope_extensions.rb [hunk 1]
```diff
@@ -278,6 +278,11 @@ def sortable_version(version)
         Util::Package.sortable_version(version)
       end
 
+      apipie :method, 'Returns true if Katello Agent infrastructure is enabled on the server'
+      def katello_agent_enabled
```
Comment (ekohl) [inline app/lib/katello/concerns/base_template_scope_extensions.rb:282]: We also have macros which follow the `?` convention. Would this be clearer?
```suggestion
      def katello_agent_enabled?
```

**Review by ekohl — APPROVED** [review]

**Review by jlsherrill — APPROVED** [review]

**Comment by jlsherrill** [comment]
don't forget to squash

File: app/lib/katello/concerns/base_template_scope_extensions.rb [hunk 2]
```diff
@@ -278,6 +278,11 @@ def sortable_version(version)
         Util::Package.sortable_version(version)
       end
 
+      apipie :method, 'Returns true if Katello Agent infrastructure is enabled on the server'
+      def katello_agent_enabled?
+        Katello.with_katello_agent?
```
Comment (jeremylenz) [inline app/lib/katello/concerns/base_template_scope_extensions.rb:283]: Should the `::` be prepended?

Comment (jturel) [reply app/lib/katello/concerns/base_template_scope_extensions.rb:283]: Works without in this case

Comment (jeremylenz) [reply app/lib/katello/concerns/base_template_scope_extensions.rb:283]: Oh, okay, uh, ACK then :yum:

**Pull request merged**